docs/arguments.rst

```diff
@@ -34,164 +38,82 @@ And what it looks like:
-    @click.argument('filename', type=click.Path(exists=True))
-    def touch(filename):
-        """Print FILENAME if the file exists."""
-        click.echo(click.format_filename(filename))
+    @click.command(context_settings={"ignore_unknown_options": True})
+    @click.argument('files', nargs=-1, type=click.Path())
+    def touch(files):
+        """Print all FILES file names."""
+        for filename in files:
+            click.echo(filename)
 
-And what it does:
+And from the command line:
 
 .. click:run::
 
-    with isolated_filesystem():
-        with open('hello.txt', 'w') as f:
-            f.write('Hello World!\n')
-        invoke(touch, args=['hello.txt'])
-        println()
-        invoke(touch, args=['missing.txt'])
-
-
-File Opening Safety
--------------------
-
-The :class:`FileType` type has one problem it needs to deal with, and that
-is to decide when to open a file.  The default behavior is to be
-"intelligent" about it.  What this means is that it will open stdin/stdout
-and files opened for reading immediately.  This will give the user direct
-feedback when a file cannot be opened, but it will only open files
-for writing the first time an IO operation is performed by automatically
-wrapping the file in a special wrapper.
-
-This behavior can be forced by passing ``lazy=True`` or ``lazy=False`` to
-the constructor.  If the file is opened lazily, it will fail its first IO
-operation by raising an :exc:`FileError`.
-
-Since files opened for writing will typically immediately empty the file,
-the lazy mode should only be disabled if the developer is absolutely sure
-that this is intended behavior.
-
-Forcing lazy mode is also very useful to avoid resource handling
-confusion.  If a file is opened in lazy mode, it will receive a
-``close_intelligently`` method that can help figure out if the file
-needs closing or not.  This is not needed for parameters, but is
-necessary for manually prompting with the :func:`prompt` function as you
-do not know if a stream like stdout was opened (which was already open
-before) or a real file that needs closing.
-
-Starting with Click 2.0, it is also possible to open files in atomic mode by
-passing ``atomic=True``.  In atomic mode, all writes go into a separate
-file in the same folder, and upon completion, the file will be moved over to
-the original location.  This is useful if a file regularly read by other
-users is modified.
+    invoke(touch, ['-foo.txt', 'bar.txt'])
 
 
 .. _environment-variables:
 
 Environment Variables
 ---------------------
 
-Like options, arguments can also grab values from an environment variable.
-Unlike options, however, this is only supported for explicitly named
-environment variables.
+This feature is not recommended since it be confusing to users. Arguments can only pull environment variables from ? explicitly named environment variables. In that case, it can also be a list of different environment variables where the first one is picked. ?
```

Comment:
@AndreasBackx Do you know what the documentation could mean in this section? I have not a clue so providing clearer wording and a better example is difficult.

Reply:
@AndreasBackx I think actually only Armin knew. It was basically unchanged since the first commit. But I figured it out by playing around. Could you give this PR a once over?